src/Rules/Drupal/RenderCallbackRule.php

```diff
@@ -166,7 +169,19 @@ private function doProcessNode(Node\Expr $node, Scope $scope, string $keyChecked
             }
 
             foreach ($typeAndMethodNames as $typeAndMethodName) {
-                if (!$trustedCallbackType->isSuperTypeOf($typeAndMethodName->getType())->yes()) {
+                $testReflection = $scope->getMethodReflection($typeAndMethodName->getType(), $typeAndMethodName->getMethod());
+                if ($testReflection === null) {
+                    $isTrustedMethod = false;
+                } else {
+                    // @todo does PHPStan have a better way?
+                    // @see https://github.com/phpstan/phpstan/discussions/5863
+                    $methodNativeClassReflection = $testReflection->getDeclaringClass()->getNativeReflection();
+                    $methodNativeReflection = $methodNativeClassReflection->getMethod($typeAndMethodName->getMethod());
+                    $isTrustedMethod = (bool) $methodNativeReflection->getAttributes('Drupal\Core\Security\Attribute\TrustedCallback');
+                }
+
+                $isTrustedCallbackType = $trustedCallbackType->isSuperTypeOf($typeAndMethodName->getType())->yes();
+                if (!$isTrustedCallbackType && !$isTrustedMethod) {
```

Comment:
This needs to be cleaned up. The discussion is from 2021, but I don't see any "better" way to reflect the method and get its attributes. Also, this should only count on Drupal >= 10.1.x. Otherwise 10.0 will see a false report about being able to use the attribute.